File: engine/engine-model/src/main/scala/tools/packet/NewYearCardResolution.scala
package tools.packet

import client.newyear.NewYearCardRecord
import net.opcodes.SendOpcode

class NewYearCardResolution(private var _characterId: Int, private var _newYearCardRecord: NewYearCardRecord,
                            private var _mode: Int, private var _message: Int) extends PacketInput {
  def characterId: Int = _characterId

  def newYearCardRecord: NewYearCardRecord = _newYearCardRecord

  def mode: Int = _mode

  def message: Int = _message

  override def opcode(): SendOpcode = SendOpcode.NEW_YEAR_CARD_RES
}

File: engine/engine-model/src/main/java/tools/packet/NewYearCardResolution.java
package tools.packet;

import client.newyear.NewYearCardRecord;
import net.opcodes.SendOpcode;

public record NewYearCardResolution(Integer characterId, NewYearCardRecord newYearCardRecord, Integer mode,
                                    Integer message) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.NEW_YEAR_CARD_RES;
   }
}